packages/lesswrong/components/sequences/TopPostsPage.tsx

```diff
@@ -814,16 +820,19 @@ function TopSpotlightsSection({classes, yearGroupsInfo, sectionsInfo, reviewWinn
 
   const filteredSpotlights = filterWhereFieldsNotNull(filteredReviewWinnersForSpotlights, 'spotlight').map(post => ({
     ...post.spotlight,
-    document: post
+    document: post,
+    ranking: post.reviewWinner?.reviewRanking
   })).sort((a, b) => {
     const reviewWinnerA = reviewWinnersWithPosts.find(post => post._id === a.document?._id)
     const reviewWinnerB = reviewWinnersWithPosts.find(post => post._id === b.document?._id)
-    if (reviewWinnerA?.reviewWinner?.reviewYear !== reviewWinnerB?.reviewWinner?.reviewYear) {  
-      return (reviewWinnerB?.reviewWinner?.reviewYear ?? 0) - (reviewWinnerA?.reviewWinner?.reviewYear ?? 0)
+    if (reviewWinnerA?.reviewWinner?.reviewRanking !== reviewWinnerB?.reviewWinner?.reviewRanking) {
+      return (reviewWinnerA?.reviewWinner?.reviewRanking ?? 0) - (reviewWinnerB?.reviewWinner?.reviewRanking ?? 0)
+    } else if (reviewWinnerA?.reviewWinner?.reviewYear !== reviewWinnerB?.reviewWinner?.reviewYear) {  
+      return (reviewWinnerA?.reviewWinner?.reviewYear ?? 0) - (reviewWinnerB?.reviewWinner?.reviewYear ?? 0)
     } else if (reviewWinnerA?.reviewWinner?.category !== reviewWinnerB?.reviewWinner?.category) {
-      return (reviewWinnerB?.reviewWinner?.category ?? '').localeCompare(reviewWinnerA?.reviewWinner?.category ?? '')
+      return (reviewWinnerA?.reviewWinner?.category ?? '').localeCompare(reviewWinnerB?.reviewWinner?.category ?? '')
     } else {
-      return (reviewWinnerB?.reviewWinner?.reviewRanking ?? 0) - (reviewWinnerA?.reviewWinner?.reviewRanking ?? 0)
+      return 0
```

Comment:
imo this would be much easier to reason about if you first handled the case whether either or both review winners were missing (i.e. the `find` returned undefined)